experimenter/queue.py

```diff
@@ -25,24 +26,79 @@ def is_running():
+    for it, worker in enumerate(workers):
+        success = worker.successful_job_count
+        fail = worker.failed_job_count
+        message = message+'\n\t\t\t worker: {}'.format(it)
+        message = message+'\n\t\t\t\t number of successful jobs: {}'.format(success)
+        message = message+'\n\t\t\t\t number of failed jobs: {}'.format(fail) 
+    return message
+    
+
+def get_failed_job(queue_name:str = _DEFAULT_QUEUE, job_num:int = 0):
+    #pass name and connection
+    reg = rq.registry.FailedJobRegistry(name = queue_name, connection = get_connection())
+    job_ids = reg.get_job_ids()
+    if (len(job_ids)<=0):
+        return "None", reg
+    job_id = job_ids[0]
+    return job_id, reg
+    
+    
+def save_failed_job(queue_name:str = _DEFAULT_QUEUE, job_num:int = 0):
+    if (queue_name is None):
+        queue_name = _DEFAULT_QUEUE
+    job_id, failed_queue = get_failed_job()
+    job = rq.job.Job.fetch(job_id, connection=get_connection())
+    with open (os.path.join('/data',"failed_job_{}.txt".format(job_num)), 'w') as job_file:
+        job_file.write(job.exc_info)
+    #remove the job
+    failed_queue.remove(job_id, delete_job=True)
+    print(_SAVE_FAILED_MESSAGE.format(os.path.join('/data',
+                              "failed_job_{}.txt".format(job_num))))
+
+
+def get_queue_message(queues: list):
+    queues_message = 'getting queues, jobs, and workers'
+    for queue in queues:
+        queues_message = queues_message + '\n\t number of jobs on queue {}: {}'.format(queue.name, len(queue))
+        _, reg = get_failed_job(queue.name)
+        num_fails = len(reg)
+        queues_message = queues_message + '\n\t number of failed jobs on queue {}: {}'.format(queue.name, num_fails)
+        workers = rq.Worker.all(queue=queue)
+        queues_message = queues_message + '\n\t\t' + str(get_worker_message(workers=workers, queue=queue))
+        
+    return queues_message
 
 
 def status() -> None:
     conn = get_connection()
-    print('available queues: {}'.format(rq.Queue.all(conn)))
-
+    queues = rq.Queue.all(conn)
+    queues_message = get_queue_message(queues)
+    print('available queues: {}'.format(queues))
+    print(queues_message)
+    
 
 def enqueue(job, queue_name: str = _DEFAULT_QUEUE, job_timeout: int = None) -> rq.job.Job:
     q = get_queue(queue_name)
     return q.enqueue(**job, job_timeout=job_timeout)
 
 
-def empty(queue_name: str = None) -> None:
+def empty(queue_name: str = None, empty_failed_queue: bool = False) -> None:
     if queue_name is None:
         queue_name = _DEFAULT_QUEUE
-    queue = get_queue(queue_name)
-    queue.empty()
-    print(_EMPTIED_MESSAGE.format(queue_name))
+    #empty the failed queue or just the normal one    
+    if (empty_failed_queue is True):
+        empty_failed(queue_name=queue_name)
+    else:
+        queue = get_queue(queue_name)
+        queue.empty()
+        print(_EMPTIED_MESSAGE.format(queue_name))
```

Comment:
Based on your experience, was it helpful to just empty one, not both at the same time?

Reply:
For me, it was helpful to just empty one at a time. This might have just been a developmental thing, and it might not be useful when things are up and running. But I do not think it would be too harmful to keep it around.